src/osd/ECCommon.cc

```diff
@@ -400,20 +435,21 @@ void ECCommon::ReadPipeline::do_read_op(ReadOp &op)
       op.obj_to_source[i->first].insert(j->first);
       op.source_to_obj[j->first].insert(i->first);
     }
-    for (list<boost::tuple<uint64_t, uint64_t, uint32_t> >::const_iterator j =
-	   i->second.to_read.begin();
-	 j != i->second.to_read.end();
-	 ++j) {
-      pair<uint64_t, uint64_t> chunk_off_len =
-	sinfo.aligned_offset_len_to_chunk(make_pair(j->get<0>(), j->get<1>()));
+    for (const auto& read : i->second.to_read) {
+      auto p = make_pair(read.offset, read.size);
+      if (g_conf()->osd_ec_partial_reads) {
+        // we need to align to stripe again to deal with partial chunk read.
+        p = sinfo.offset_len_to_stripe_bounds(p);
+      }
+      pair<uint64_t, uint64_t> chunk_off_len = sinfo.aligned_offset_len_to_chunk(p);
```

Comment:
> if !g_conf()->osd_ec_partial_reads then p is already stripe aligned though it might be stripe aligned anyway if this happens to be a fast read.

Yes, that's also mine understanding.

> Generally speaking, I'd like ReadPipeline's behavior to generalize to handling partial reads without needing special handling as much as possible. Otherwise, the testing burden will be much higher.

There are some places also in `ECBacked` driving assumptions on the stripe alignment. I'm trying to rework them,